sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -26,15 +26,23 @@ public void Execute(GeneratorExecutionContext context)
 
             // attempt to parse user compilation
             var p = new Parser(context);
-            IReadOnlyList<GeneratorFunctionMetadata> functionMetadataInfo = p.GetFunctionMetadataInfo(receiver.CandidateMethods);
 
-            // Proceed to generate the file if function metadata info was successfully returned
-            if (functionMetadataInfo.Count > 0)
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.FunctionsDisableFuncMetadataSourceGen", out var sourceGenSwitch);
```

Comment:
Small suggestion: can we start a source gen property naming format? So that if we introduce other build props to control source gen later on, they will all follow the same naming. I'd recommend something like: `FunctionsMetadataSourceGen_{Property}`, so `FunctionsMetadataSourceGen_Enabled` (and have it default to `true` if not present).

The choice for "Enabled" is also a different preference of mine: I like to name bools in the _positive_ term (I do make occasional exceptions though).

Reply:
@jviau I was thinking that if we plan to make source-gen the default, then the action should be to disable it. Positive term is definitely more intuitive though - changed it to that.